src/closures.rs

```diff
@@ -91,8 +91,9 @@ fn get_inner_expr<'a>(
 ) -> &'a ast::Expr {
     if let ast::ExprKind::Block(ref block, _) = expr.kind {
         if !needs_block(block, prefix, context) {
-            // block.stmts.len() == 1
-            if let Some(expr) = stmt_expr(&block.stmts[0]) {
+            // block.stmts.len() == 1 except with `|| {{}}`;
+            // https://github.com/rust-lang/rustfmt/issues/3844
+            if let Some(Some(expr)) = block.stmts.first().map(stmt_expr) {
```

Comment:
My bad...we should use `and_then` instead of `map` in this case.
```suggestion
            if let Some(expr) = block.stmts.first().and_then(stmt_expr) {
```

Reply:
No worries!